pkg/resources/task.go

```diff
@@ -191,110 +142,73 @@ func Task() *schema.Resource {
 		d.SetId("")
 		return nil
 	}
-	if err != nil {
-		return err
-	}
 
-	if err := d.Set("enabled", t.IsEnabled()); err != nil {
+	if err := d.Set("enabled", task.IsStarted()); err != nil {
 		return err
 	}
 
-	if err := d.Set("name", t.Name); err != nil {
+	if err := d.Set("name", task.Name); err != nil {
 		return err
 	}
 
-	if err := d.Set("database", t.DatabaseName); err != nil {
+	if err := d.Set("database", task.DatabaseName); err != nil {
 		return err
 	}
 
-	if err := d.Set("schema", t.SchemaName); err != nil {
+	if err := d.Set("schema", task.SchemaName); err != nil {
 		return err
 	}
 
-	if err := d.Set("warehouse", t.Warehouse); err != nil {
+	if err := d.Set("warehouse", task.Warehouse); err != nil {
 		return err
 	}
 
-	if err := d.Set("schedule", t.Schedule); err != nil {
+	if err := d.Set("schedule", task.Schedule); err != nil {
 		return err
 	}
 
-	if err := d.Set("comment", t.Comment); err != nil {
+	if err := d.Set("comment", task.Comment); err != nil {
 		return err
 	}
 
-	allowOverlappingExecutionValue, err := t.AllowOverlappingExecution.Value()
-	if err != nil {
+	if err := d.Set("allow_overlapping_execution", task.AllowOverlappingExecution); err != nil {
 		return err
 	}
 
-	if allowOverlappingExecutionValue != nil && allowOverlappingExecutionValue.(string) != "null" {
-		allowOverlappingExecution, err := strconv.ParseBool(allowOverlappingExecutionValue.(string))
-		if err != nil {
-			return err
-		}
-
-		if err := d.Set("allow_overlapping_execution", allowOverlappingExecution); err != nil {
-			return err
-		}
-	} else {
-		if err := d.Set("allow_overlapping_execution", false); err != nil {
-			return err
-		}
-	}
-
-	// The "DESCRIBE TASK ..." command returns the string "null" for error_integration
-	if t.ErrorIntegration.String == "null" {
-		t.ErrorIntegration.Valid = false
-		t.ErrorIntegration.String = ""
-	}
-
-	if err := d.Set("error_integration", t.ErrorIntegration.String); err != nil {
+	if err := d.Set("error_integration", task.ErrorIntegration); err != nil {
 		return err
 	}
 
-	predecessors, err := t.GetPredecessors()
-	if err != nil {
-		return err
+	predecessors := make([]string, len(task.Predecessors))
+	for i, p := range task.Predecessors {
+		predecessors[i] = p.Name()
 	}
-
 	if err := d.Set("after", predecessors); err != nil {
 		return err
 	}
 
-	if err := d.Set("when", t.Condition); err != nil {
+	if err := d.Set("when", task.Condition); err != nil {
 		return err
 	}
 
-	if err := d.Set("sql_statement", t.Definition); err != nil {
+	if err := d.Set("sql_statement", task.Definition); err != nil {
 		return err
 	}
 
-	q = builder.ShowParameters()
-	paramRows, err := snowflake.Query(db, q)
-	if err != nil {
-		return err
-	}
-	params, err := snowflake.ScanTaskParameters(paramRows)
+	opts := &sdk.ShowParametersOptions{In: &sdk.ParametersIn{Task: taskId}}
+	params, err := client.Parameters.ShowParameters(ctx, opts)
 	if err != nil {
 		return err
 	}
```

Comment:
log.Printf("[TRACE] %+v\n", param)
maybe lets not print the param

Reply:
Okay, that was part of the previous implementation - removed.